java/code/src/com/redhat/rhn/common/security/acl/AclFactory.java

```diff
@@ -54,14 +46,15 @@ public static AclFactory getInstance() {
      */
     public Acl getAcl(String mixinsIn) {
         Acl aclObj = new Acl();
-        Access access = new Access();
         aclObj.registerHandler(access);
 
         // Add the mixin handlers as well.
         if (mixinsIn != null) {
             String[] mixin = StringUtils.split(mixinsIn, ",");
             for (int i = 0; i < mixin.length; i++) {
-                aclObj.registerHandler(StringUtils.trim(mixin[i]));
+                if (!mixin[i].equals(Access.class.getName())) {
```

Comment:
yeah the method name is a bit weird. There is also a todo comment at the top of this class about implementing cashing. I think this would be a good topic for another refactoring while also getting rid of the use of reflection here.